Cassette/CASQueueFile.m

```diff
@@ -586,14 +612,16 @@ - (NSUInteger)wrapPosition:(NSUInteger)position {
 }
 
 /**
- * If necessary, expands the file to accommodate an additional element of the
- * given length.
+ * If necessary, expands the file to accommodate the given elements.
  *
  * Returns YES on success. On failure, returns NO and sets *error.
  */
-- (BOOL)expandIfNecessary:(NSUInteger)elementLength
+- (BOOL)expandIfNecessary:(NSArray<NSData *> *)elements
                     error:(NSError * __autoreleasing * _Nullable)error {
-    NSUInteger numBytesRequested = ElementHeaderLength + elementLength;
+    NSUInteger numBytesRequested = ElementHeaderLength;
```

Comment:
You are totally right, I undercounted. However, I couldn't find where we write a separate header for each element — can you check this math? It seems like the code writes:

1) A single header of `ElementHeaderLength` (16 bytes), then for each element:
2) A `uint32_t` (4 bytes) for the length of the element, then
3) The bytes of the serialized element

I think using `ElementHeaderLength * elements.count` will work, but I think it'll cause the file to be larger than it needs to be by quite a lot — for 4k elements, that'd be 64k of overhead rather than 16k of overhead.

Reply:
Ohh, I see where I messed up — `ElementHeaderLength` is indeed 4 bytes. I mistakenly thought this was `QueueFileHeaderLength` (16 bytes). Thanks for the sharp eyes!